app_feup/lib/model/entities/lecture.dart

```diff
@@ -40,17 +40,43 @@ class Lecture {
         ((startTimeSeconds % 3600) ~/ 60).toString().padLeft(2, '0');
   }
 
+  Lecture.secConstructor(String subject, String typeClass, int day,
```

Comment:
I used `secConstructor` since this was an already existing name used for a secondary constructor: https://github.com/NIAEFEUP/project-schrodinger/blob/00fb87f3ca65d96c33dfcb4f35d4fa0e01f0f854/app_feup/lib/model/entities/exam.dart

Reply:
For consistency, I think it was a good choice!